python/rmm/_lib/device_buffer.pyx

```diff
@@ -285,6 +285,58 @@ cdef class DeviceBuffer:
 
         return b
 
+    cpdef DeviceBuffer join(self, list L, Stream stream=DEFAULT_STREAM):
+        """Joins a sequence of ``DeviceBuffer``s with ``self`` inbetween.
```

Comment:
I think this would be clearer if you actually copied more of the docstring from `help(b''.join)`. Specifically, it's not obvious that "in between" means that `self` is inserted between _every_ consecutive element. Use of the word "separator" to describe `self` might also help.